api/graph.py

```diff
@@ -416,18 +416,12 @@ def delete_files(self, files: list[dict]) -> tuple[str, dict, list[int]]:
 
         q = """UNWIND $files AS file
                MATCH (f:File {path: file['path'], name: file['name'], ext: file['ext']})
-               WITH collect(f) AS Fs
-               UNWIND Fs AS f
-               OPTIONAL MATCH (f)-[:DEFINES]->(e)
-               WITH Fs, collect(e) AS Es
-               WITH Fs + Es AS entities
-               UNWIND entities AS e
-               DELETE e
-               RETURN collect(ID(e))
+               OPTIONAL MATCH (f)-[:DEFINES*]->(e)
+               DELETE f, e
         """
 
-        params = {'files': files}
-        res = self._query(q, params)
+        params = {'files': [{'path': str(file_path), 'name': file_path.name, 'ext' : file_path.suffix} for file_path in files]}
```

Comment:
Please verify that the index is utilized I have a feeling that the `MATCH` above doesn't utilize the index, this is an upcoming change I'll introduce to the DB.

Reply:
there is an index of file name and ext

Reply:
the fact that there's an index is one thing, I want to make sure the index is indeed utilized by this query.